packages/vaex-ml/vaex/ml/transformations.py

```diff
@@ -415,10 +415,16 @@ def fit(self, df):
         :param df: A vaex DataFrame.
         '''
 
-        assert len(self.feature_range) == 2, 'feature_range must have 2 elements only'
-        minmax = df.minmax(self.features)
-        self.fmin_ = minmax[:, 0].tolist()
-        self.fmax_ = minmax[:, 1].tolist()
+        fmin = df.min(self.features, delay=True)
+        fmax = df.max(self.features, delay=True)
```

Comment:
Hmm, i am a bit worried that this passes the tests. This means it's being cast to a common type.

Reply:
I think we should loop over each feature, we can still call minmax right? (which is faster than a call to min, and to max)

Reply:
Both options would use 1 pass over the data right? Why would it be faster?
I can try to make a loop if you want, or if you have time/will you can also implement it.

Reply:
There is a difference right. If I do a df.min(['somefloat', 'someint']), it will come back as float array, we don't want that.

Reply:
Done. Now each feature preserves the dtype of its min/max values in MinMaxScaler